extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -51,47 +51,58 @@ const metadataProvider = classes.MetadataProvider;
  * @param {string|bool} singlepart - indicates of the retrieves can fetch singlepart.  Options are bulkdata, video, image or boolean true
  */
 function createDicomWebApi(dicomWebConfig, userAuthenticationService) {
-  const {
-    qidoRoot,
-    wadoRoot,
-    enableStudyLazyLoad,
-    supportsFuzzyMatching,
-    supportsWildcard,
-    supportsReject,
-    staticWado,
-    singlepart,
-  } = dicomWebConfig;
-
-  const dicomWebConfigCopy = JSON.parse(JSON.stringify(dicomWebConfig));
-
-  const qidoConfig = {
-    url: qidoRoot,
-    staticWado,
-    singlepart,
-    headers: userAuthenticationService.getAuthorizationHeader(),
-    errorInterceptor: errorHandler.getHTTPErrorHandler(),
-  };
+  let dicomWebConfigCopy,
+    qidoConfig,
+    wadoConfig,
+    qidoDicomWebClient,
+    wadoDicomWebClient,
+    isInitialized = false;
 
-  const wadoConfig = {
-    url: wadoRoot,
-    staticWado,
-    singlepart,
-    headers: userAuthenticationService.getAuthorizationHeader(),
-    errorInterceptor: errorHandler.getHTTPErrorHandler(),
-  };
+  const implementation = {
+    initialize: ({ params, query }) => {
```

Comment:
This method should be able to grab a new config too to override the old one

Reply:
Updated the code so the initialize is called once as it should be.

Reply:
I think @sedghi's suggestion above still needs to be addressed, although I am not sure if this PR is necessarily the best place to address it. The suggestion stems from a community request for being able to dynamically change/update the data source configuration via a `postMessage` from a parent/ancestor when OHIF is embedded in an iframe. I think it is going to be tricky because the `postMessage` could arrive at any arbitrary time.

Reply:
Added the option to override the config 👍 One could call initialize again using postMessage.

Reply:
Thanks for this. I suppose if ever `initialize` is called with a `config`, it would be up to the caller to do "something" so that the new config is applied "everywhere" that the former had already been applied to... perhaps even a page/router refresh... but you don't have to worry about that... I just got lost in thought... lol. Thanks again.

Reply:
Actually I was struggling with the `postMessage` implementation for override the config with new one. Can you please suggest step to override config using this PR?

Reply:
@anarnoli , I know we have spoken outside of this PR, but I just wanted to record what we spoke about here for completeness...

- In the `preRegistration` method of your extension you should be able to grab an instance of `extensionManager `.
- `extensionManager` has a `getDataSources` method.
- Loop through the data sources returned, looking for the data source you want to update the configuration for.
- Once you have the data source to (re)configure, call the new `initialize` method (of this PR) on it with the new configuration.
- Then the question I have is that the above process can happen at ANY point in time. For example, a study might already have been loaded and then the `postMessage` arrives. We need to add some thought about how to apply the new configuration. One way might be to use the React router to simply refresh the page. Not sure that will work, but worth a shot.